backend/app.py

```
@@ -0,0 +1,375 @@
+    allowed_categories = {"Music", "Tech", "Sports", "Art", "Dance"}
+    if data["category"] not in allowed_categories:
+        return jsonify({"message": f"Category must be one of: {', '.join(sorted(allowed_categories))}"}), 400
+
+    # Validate coordinates
+    try:
+        lat = float(data["lat"])
+        lng = float(data["lng"])
+        if not (-90 <= lat <= 90) or not (-180 <= lng <= 180):
+            raise ValueError
+    except (ValueError, TypeError):
+        return jsonify({"message": "lat must be −90..90 and lng must be −180..180"}), 400
```

Ruff (0.15.1)

[warning] String contains ambiguous `−` (MINUS SIGN). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)